Big cAVS tooling update

andyross opened the pull request:
The confusion about different scripting environments and how to do testing using tools like twister was getting a little out of control.  This steals the cavstool.py patch out of the IPC PR and updates the older twister integration (now named "cavstwist.sh") in the same directory. 

And it kills all the older scripts in the process.

Commit:
soc/intel_adsp: Add unified "cavstool" loader/logger script

The existing scripting for these platforms has gotten a little stale.
The loader had bifurcated into a v15 and v25 variant, both of which
lived in the cavs15 board directory.  Building off Shao Ming's
excellent (if somewhat surprisingly committed) rework to unify
unchanged parts of the scripts, let's finish the job.

This adds a "cavstool.py" script with the following advantages:

+ It's just one script for everything, with a single unified load
  process that works reliably on both 1.5 and 1.8+ hardware.

+ It runs on all cAVS platforms (with a compatible kernel, those
  requirements haven't changed)

+ It automatically emitts logging synchronously after loading,
  eliminating the race between adsplog.py and cavs-fw.py where you
  could see logging from a previous test run.

+ It automatically detects and unloads a linux kernel module managing
  the same device (even if SOF has renamed the module again, heh).

+ Timings have been tuned up in general, it's about 2 seconds faster
  to get to first log output now.

Signed-off-by: Andy Ross <andrew.j.ross@intel.com>

Commit:
soc/intel_adsp: Add twister integration script

This is based closely on the older cavsload.sh script, but updated to
keep up with interesting new twister behavior, with a cleaner
interface for CI integration that doesn't require editing the file,
somewhat better internal documentation, and with a more focused eye on
use as a twister tool specifically.

Signed-off-by: Andy Ross <andrew.j.ross@intel.com>

Commit:
boards/intel_adsp_cavs15: Remove ancient tooling

There were several generations of loader/logger tooling in this
directory, several of which no longer work.  They have all been
replaced by just one pythong script in the soc directory.

Signed-off-by: Andy Ross <andrew.j.ross@intel.com>

teburd commented on soc/xtensa/intel_adsp/tools/cavstool.py:
```diff
@@ -0,0 +1,300 @@
+    (hda, sd, dsp, hda_ostream_id, cavs15) = map_regs()
+    log.info(f"Detected cAVS {'1.5' if cavs15 else '1.8+'} hardware")
+
+    load_firmware(sys.argv[1])
```
Allow for logging only perhaps, though perilous....

```suggestion
    if len(sys.argv) > 1:
        load_firmware(sys.argv[1])
```

teburd commented on soc/xtensa/intel_adsp/tools/cavstwist.sh:
```diff
@@ -0,0 +1,110 @@
+#
+#  export CAVS_HOST=tgl2
+#  export CAVS_KEY=$HOME/otc_private_key_3k.pem
+#  export CAVS_RIMAGE=$HOME/rimage
```
I'd add a note/comment here that this is the path to a directory containing rimage and the config directory, this bit me and I had to look harder at what was going on.

teburd reviewed (changes requested):
much much nicer than my own cobbled scripts from the cavsload.sh that existed before, seems to work quite well for me

Really +1 with some minor nits to help the next person that does what I did

Commit:
soc/intel_adsp: Polish cavstool

There's desire for a "log only" mode like the older adsplog script
had.  Add a few other quality of life command line options too.  And
catch the most obvious user errors to print a message instead of a
stack dump.

Signed-off-by: Andy Ross <andrew.j.ross@intel.com>

teburd reviewed (approved):
I'm able to reliably do twister runs without fail now with this

andyross commented on boards/xtensa/intel_adsp_cavs15/tools/adsplog.py:
```diff
@@ -1,104 +0,0 @@
-#!/usr/bin/python3
```
cavstool -l, see last patch

nashif replied:
oh sorry, those were comments I wrote earlier and did not submit....

andyross commented on soc/xtensa/intel_adsp/tools/cavstwist.sh:
```diff
@@ -0,0 +1,110 @@
+
+    ELF=$BLDDIR/zephyr/zephyr.elf.mod
+    BOOT=$BLDDIR/zephyr/bootloader.elf.mod
+    (cd $BLDDIR; west sign --tool-data=$CAVS_RIMAGE/config -t rimage -- -k $CAVS_KEY) >/dev/null 2>&1
```
fixed
